blockchain/tx_list.go

```diff
@@ -658,7 +658,8 @@ func (l *txPricedList) Underpriced(tx *types.Transaction, local *accountSet) boo
 
 // Discard finds a number of most underpriced transactions, removes them from the
 // priced list and returns them for further removal from the entire pool.
-func (l *txPricedList) Discard(slots int, local *accountSet) types.Transactions {
+// If noPending is set to true, we will only consider the floating list
+func (l *txPricedList) Discard(slots int, local *accountSet) (types.Transactions, bool) {
```

Comment:
Is the returned bool value determined solely based on the length of `drop`? Specifically, does it return true only when `drop` is empty, and false otherwise?
And just to confirm, this return value is unrelated to the `save` list, right?

Reply:
This is a good point. We need to address them in this PR, as we are discussing [here](https://github.com/kaiachain/kaia/pull/391#pullrequestreview-2877050046). I only applied before [Geth PR](https://github.com/ethereum/go-ethereum/pull/21478)

> Is the returned bool value determined solely based on the length of drop?
Yes, it does.

> Specifically, does it return true only when drop is empty, and false otherwise?

I don't think so. If "slots -= numSlots(tx)" becomes zero, then it returns true (drop is empty or not)

> And just to confirm, this return value is unrelated to the save list, right?

Yes, it does. But I'm sure that it's not correct.

I say again, we have to address them in this PR. Let me apply them.

Reply:
okay thanks!

Reply:
>> And just to confirm, this return value is unrelated to the save list, right?

> Yes, it does

Because Kaia is discarding remote txs only now.